src/logger/src/metrics.rs

```diff
@@ -24,10 +56,100 @@
 //! If if turns out this approach is not really what we want, it's pretty easy to resort to
 //! something else, while working behind the same interface.
 
-use std::sync::atomic::{AtomicUsize, Ordering};
+use std::io::Write;
+use std::ops::Deref;
+use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+use std::sync::Mutex;
 
 use serde::{Serialize, Serializer};
 
+use error::MetricsError;
+
+use super::{buf_guard, write_to_destination};
+
+lazy_static! {
+    /// Static instance used for handling metrics.
+    pub static ref METRICS: Metrics<FirecrackerMetrics> = Metrics::new(FirecrackerMetrics::default());
+}
+
+/// Metrics system.
+// All member fields have types which are Sync, and exhibit interior mutability, so
+// we can call operations on metrics using a non-mut static global variable.
+pub struct Metrics<T: Serialize> {
+    // Metrics will get flushed here.
+    metrics_buf: Mutex<Option<Box<dyn Write + Send>>>,
+    is_initialized: AtomicBool,
+    pub app_metrics: T,
+}
+
+impl<T: Serialize> Metrics<T> {
+    /// Creates a new instance of the current metrics.
+    // TODO: We need a better name than app_metrics (something that says that these are the actual
+    // values that we are writing to the metrics_buf).
+    pub fn new(app_metrics: T) -> Metrics<T> {
```

Comment:
It might seem basic, but I think it is pretty descriptive and sufficient: 

```suggestion
    pub fn new(metrics: T) -> Metrics<T> {
```

Also, the current `app_metrics` seems good to me.

Reply:
Hmm, `metrics` seems too general in this particular situation. Maybe `metrics_values` or `metrics_to_flush`, I don't know  :-??.